Events/BodyScannedEvent.cs

```diff
@@ -136,5 +140,84 @@ private decimal sanitiseCP(decimal cp)
                 return Math.Round(cp * 100);
             }
         }
+
+        private long? estimatevalue(bool? dssequipped)
+        {
+            int bodydataconstant = 720;
+            int terraconstant = 0;
+            bool terraformable = false;
+            double value;
+            double basevalue;
+            double terrabonusvalue;
+
+            // Override constants for specific types of bodies
+            if ( (terraformstate == "Terraformable") || (terraformstate == "Candidate for terraforming") )
+            {
+                terraformable = true;
+            }
+            if ( bodyclass == "Ammonia world")
+            {
+                // Ammonia worlds
+                bodydataconstant = 232619;
+            }
+            else if ( bodyclass == "Earthlike body" )
+            {
+                // Earth-like worlds
+                bodydataconstant = 155581;
+                terraconstant = 279088;
+            }
+            else if (bodyclass == "Water world" )
+            {
+                // Water worlds
+                bodydataconstant = 155581;
+                if (terraformable)
+                {
+                    terraconstant = 279088;
+                }
+            }
+            else if (bodyclass == "Metal rich body")
+            {
+                // Metal rich worlds
+                bodydataconstant = 52292;
+            }
+            else if (bodyclass == "High metal content body")
+            {
+                // High metal content worlds
+                bodydataconstant = 23168;
+                if (terraformable)
+                {
+                    terraconstant = 241607;
+                }
+            }
+            else if (bodyclass == "Rocky body")
+            {
+                // Rocky worlds
+                if (terraformable)
+                {
+                    terraconstant = 223971;
+                }
+            }
+            else if (bodyclass == "Sudarsky class I gas giant")
+            {
+                // Class I gas giants
+                bodydataconstant = 3974;
+            }
+            else if (bodyclass == "Sudarsky class II gas giant")
+            {
+                // Class II gas giants
+                bodydataconstant = 23168;
+            }
+
+            // Calculate exploration scan values
+            basevalue = bodydataconstant + (3 * bodydataconstant * Math.Pow((double)earthmass, 0.199977) / 5.3);
```

Comment:
Probably a good idea to link to documentation for these formulae.

Reply:
The documentation is the source code for Jackie Silver's habitable zone calculator (used with permission, requested via Discord) and the source code for EDDiscovery (which uses the same formulas from Jackie Silver). 
Unfortunately, I don't have anything more clearly written that I can reference at the moment.